src/node_report.h

```diff
@@ -13,6 +13,7 @@
 #include <unistd.h>
 #endif
 
+#include <filesystem>
```

Comment:
It seems wrong to include a header without using it in the file. Shouldn't it be included in `node_report.cc` ?